### ISSUE TITLE
GIT_SHALLOW still downloads the whole repo

### ISSUE THREAD

**Issue by Leafmun-certii** [cpm-cmake/CPM.cmake#540]
I am using CPMAddPackage to download a repo at a tag containing a lot less than the main branch. I have specified GIT_SHALLOW, however, this downloads the whole repo before switching to the tag I need. 

Is there any way to just download the files specified at the tag? This might be an issue with CMake's [Fetch Content ](https://discourse.cmake.org/t/shallow-downloads-of-git-repos-via-fetchcontent/10220) but I thought I'd ask here as well just in case anyone has any bright ideas.

Thanks all!

**Comment by Longwater1234** [comment]
@Leafmun-certii  Yes this is a common issue. You may want to use GIT_TAG (with hash of the stable commit) instead.
